cms/static/js/views/paged_container.js

```diff
@@ -112,23 +125,48 @@ define(["jquery", "underscore", "js/views/container", "js/utils/module", "gettex
                 this.pagingFooter.render();
             },
 
-            refresh: function(block_added) {
-                if (block_added) {
-                    this.collection.totalCount += 1;
-                    this.collection._size +=1;
-                    if (this.collection.totalCount == 1) {
-                        this.render();
+            refresh: function(xblockView, block_added, is_duplicate) {
+                if (!block_added) {
+                    return
+                }
+                if (is_duplicate) {
+                    // Duplicated blocks can be inserted onto the current page.
+                    var xblock = xblockView.xblock.element.parents(".studio-xblock-wrapper").first();
+                    var all_xblocks = xblock.parent().children(".studio-xblock-wrapper");
+                    var index = all_xblocks.index(xblock);
+                    if ((index + 1 <= this.page_size) && (all_xblocks.length > this.page_size)) {
+                        // Pop the last XBlock off the bottom.
+                        all_xblocks[all_xblocks.length - 1].remove();
                         return
                     }
-                    this.collection.totalPages = this.getPageCount(this.collection.totalCount);
-                    var new_page = this.collection.totalPages - 1;
-                    // If we're on a new page due to overflow, or this is the first item, set the page.
-                    if (((this.collection.currentPage) != new_page) || this.collection.totalCount == 1) {
-                        this.setPage(new_page);
+                }
+                this.collection.totalCount += 1;
+                this.collection._size +=1;
+                if (this.collection.totalCount == 1) {
+                    this.render();
+                    return
+                }
+                this.collection.totalPages = this.getPageCount(this.collection.totalCount);
+                var new_page = this.collection.totalPages - 1;
+                // If we're on a new page due to overflow, or this is the first item, set the page.
+                if (((this.collection.currentPage) != new_page) || this.collection.totalCount == 1) {
+                    var force_render = JSON.stringify([xblockView.model.id]);
```

Comment:
I think it would be better to just be a string for the ID. When I read the Python, I was confused why it was a list. I thought the JS might make this clear, but then I was still confused. :)